torch_brain/utils/readout.py

```diff
@@ -16,40 +16,45 @@ def prepare_for_readout(
     data: Data,
     readout_spec: "ModalitySpec",
 ):
-    required_keys = ["readout_id"]
-    optional_keys = [
-        "weights",
-        "normalize_mean",
-        "normalize_std",
-        "timestamp_key",
-        "value_key",
-        "metrics",
-        "eval_interval",
-    ]
-
-    readout_config = data.config["readout"]
-
-    # check that the readout config contains all required keys
-    for key in required_keys:
-        if key not in readout_config:
-            raise ValueError(f"readout config is missing required key: {key}")
-
-    # check that the readout config contains only valid keys
-    if not all(key in required_keys + optional_keys for key in readout_config.keys()):
-        raise ValueError(
-            f"Readout {readout_config} contains invalid keys, please use only {required_keys + optional_keys}"
-        )
-
-    key = readout_config["readout_id"]
-
-    if key not in torch_brain.MODALITY_REGISTRY:
-        raise ValueError(
-            f"Readout {key} not found in modality registry, please register it "
-            "using torch_brain.register_modality()"
-        )
-
-    value_key = readout_config.get("value_key", readout_spec.value_key)
-    timestamp_key = readout_config.get("timestamp_key", readout_spec.timestamp_key)
+
+    readout_config = dict(readout_spec)
+
+    if "readout" in data.config:
+        # Ensure readout config contains valid keys and merge it with the
+        # default config provided by readout_spec
+
+        _readout_config = data.config["readout"]
+
+        required_keys = ["readout_id"]
```

Comment:
Note: I am aware that `readout_id` is completely redundant here (because `readout_spec` is a parameter of this function). I'm keeping it here, just so that the readout config format of `prepare_for_readout` and `prepare_for_multitask_readout` remain similar.